pdns/pdns_recursor.cc

```diff
@@ -5273,6 +5338,9 @@ int main(int argc, char **argv)
     ::arg().set("unique-response-db-size", "Size of the DB used to track unique responses in terms of number of cells. Defaults to 67108864")="67108864";
     ::arg().set("unique-response-pb-tag", "If protobuf is configured, the tag to use for messages containing unique DNS responses. Defaults to 'pdns-udr'")="pdns-udr";
 #endif /* NOD_ENABLED */
+
+    ::arg().setSwitch("extended-errors", "If set, send the EDNS Extended Error extension on DNSSEC validation failures")="no";
```

Comment:
Right, I think you raise a real issue. We need to decide what level of granularity we want there, because we already have:
- extended errors on validation ;
- extended errors from RPZ (which is already configurable at the RPZ level, but still requires `extended-errors` to be enabled at the moment ;
- and extended errors from Lua (which is the same).

Since the last two require an additional configuration step, I would be inclined to rename `extended-errors` to `extended-errors-on-validation` as you suggested, making extended errors from RPZ and Lua _NOT_ dependent on that setting. Do you agree?

Reply:
On second thought, what if we decide to generate extended errors on network error, unreachable authority, since these codes do exist already? Would we add a new setting? Or would it not make sense to also have that behind `extended-errors`?
Perhaps I should only make  extended errors from RPZ and Lua independent of that setting?

Reply:
I thin there are three groups:
- EEs coming from auths
- EEs generated by rec itself on some conditions
- EEs set by the user via RPZ or Lua

The last ones do not need a switch, since if you do not want them just do net set the RPZ option or leave the Lua code out. The first two should be separate options I think. Or maybe one option with multiple values: off, internal, external, all?

Reply:
So I renamed the option to `extended-resolution-errors`, that way we can extend the current errors to include network error, unreachable authority, etc. EDE generated from `RPZ` or `Lua` are sent without having to set that `on`. Does that work for you?

Reply:
yes